korangar/src/graphics/passes/forward/shader/entity.wgsl

```diff
@@ -205,16 +205,40 @@ fn fs_main(input: VertexOutput) -> FragmentOutput {
             visibility = f32(mapped_distance - bias < shadow_map_depth);
         }
 
-        let attenuation = calculate_attenuation(light_distance, light.range) * visibility;
-        final_color += light_percent * light.color.rgb * diffuse_color.rgb * attenuation;
+        let intensity = 10.0;
+        let attenuation = calculate_attenuation(light_distance, light.range);
+        point_light_contribution += (light.color.rgb * intensity) * light_percent * attenuation * visibility;
+    }
+
+    let base_color = diffuse_color.rgb * input.color.rgb;
+    let light_contributions = saturate(ambient_light_contribution + directional_light_contribution + point_light_contribution);
+    var color = base_color.rgb * light_contributions;
+
+    if (global_uniforms.enhanced_lightning == 0) {
+        color = color_balance(color, -0.01, 0.0, 0.0);
     }
 
     var output: FragmentOutput;
-    output.fragment_color = vec4<f32>(final_color, final_alpha);
+    output.fragment_color = vec4<f32>(color, alpha_channel);
```

Comment:
Remember to multiply alpha_channel by input.color.a

Reply:
Nice catch!